samples/algorithms/Measurement.qs

```diff
@@ -0,0 +1,37 @@
+/// # Sample
+/// Measurement
+///
+/// # Description
+/// Quantum measurement is an irreversible operation in which a quantum system
+/// is manipulated to yield a numerical result. Measuring a quantum system
+/// generally changes the quantum state that describes that system.
+///
+/// In Q#, the result of a measurement is a value of the type `Result`, that is,
+/// `One` or `Zero`.
+///
+/// This Q# program exemplifies different types of measurements.
+namespace Sample {
+    open Microsoft.Quantum.Measurement;
+    @EntryPoint()
+    operation Main () : (Result, Result[], Result) {
+        // The `M` operation performs a measurement of a single qubit in the
+        // computational basis, also known as the Pauli Z basis.
+        use q = Qubit();
+        let result = M(q);
+        Reset(q);
+
+        // The `MeasureEachZ` operation measures each qubit in an array in the
+        // computational basis and returns an array of `Result` values.
+        use qs = Qubit[2];
+        let results = MeasureEachZ(qs);
+
+        // The `Measure` operation performs a joint measurement of one or more
```

Comment:
This seems like it should be reserved for the Joint Measurement sample. Maybe just show how `Measure` can be used to measure a single qubit in a different base that `PauliZ`, and maybe put a reference to the Joint Measurement sample as a "see more on Joint Measurement".

Reply:
I think probably removing `Measure` from here is a good since each sample should be self-contained and we should not reference other samples.